shared/src/main/java/com/example/shared/model/Game.java

```diff
@@ -248,60 +244,54 @@ public void sendDeckCount(){
     }
 
     public void cardSelected(String username, int index) {
-        //5 is the index for the face-down-deck
-        for (Player player : playerList) {
-            if (player.getUsername().equals(username)) {
-                if (index != 5) {
-                    player.addCard(trainCardsFaceUp.get(index));
-                    trainCardsFaceUp.set(index, trainCardDeck.drawFromTop());
-                    clientProxy.updateFaceUpCards(trainCardsFaceUp);
-                    clientProxy.sendDeckCount(ticketDeck.getDeckSize(), trainCardDeck.getDeckSize());
-                } else {
-                    TrainCard newCard = trainCardDeck.drawFromTop();
-                    player.addCard(newCard);
-                    clientProxy.receiveFaceDownCard(newCard, username, gameId);
-                    clientProxy.sendDeckCount(ticketDeck.getDeckSize(), trainCardDeck.getDeckSize());
-                }
-            }
+        if (index != 5) {
+            getPlayerWithUsername(username).addTrainCard(trainCardsFaceUp.get(index));
+            trainCardsFaceUp.set(index, trainCardDeck.drawFromTop());
+            clientProxy.updateFaceUpCards(trainCardsFaceUp);
+            clientProxy.sendDeckCount(ticketDeck.getDeckSize(), trainCardDeck.getDeckSize());
+        }
+
+        else { //5 is the index for the face-down-deck
+            TrainCard newCard = trainCardDeck.drawFromTop();
+            getPlayerWithUsername(username).addTrainCard(newCard);
+            clientProxy.receiveFaceDownCard(newCard, username, gameId);
+            clientProxy.sendDeckCount(ticketDeck.getDeckSize(), trainCardDeck.getDeckSize());
         }
     }
 
     public void ticketsRequested(String username) {
-        clientProxy.ticketsReceived(this.initializeTickets(), username, gameId);
+        clientProxy.ticketsReceived(this.initializeTickets(username), username, gameId);
     }
 
-    public void ticketsReturned(String username, ArrayList<Ticket> returned) {
-        for (Player player : playerList) {
-            if (player.getUsername().equals(username)) {
-                List<Ticket> tickets = player.getTickets();
-                for (Ticket ticket : returned) {
-                    if (tickets.contains(ticket)) {
-                        tickets.remove(ticket);
-                    }
-                }
-            }
+    public void ticketsReturned(String gameId, String username, ArrayList<Ticket> returned) {
```

Comment:
Why would you pass in the gameId if the game object already has it as a field?

Reply:
I think it's a habit we got into. It's my fault I think.